staff/oriol-bover/horroy/scripts/horroy.js

```diff
@@ -0,0 +1,328 @@
+function Horroy() {
+    this.length = arguments.length;
+
+    if (arguments.length)
+        for (var i = 0; i < arguments.length; i++)
+            // this.push(arguments[i]); // WARN should avoid (if possible) calling member methods in a constructor (push)
+            this[i] = arguments[i];
+}
+
+/**
+ *  To String
+ * 
+ * Converts an horroy object to String. Their elements will be printed seperated by commas.
+ * 
+ * @returns {String} 
+ * 
+ */
+
+Horroy.prototype.toString = function() {
+    var string = '';
+    
+    for (var i = 0; i < this.length; i++) {
+        if (i === this.length-1) {
+            string += this[i]
+        } else {
+            string += this[i] + ','
+        }
+    }
+    return string
+};
+
+
+/**
+ * 
+ * From
+ * 
+ * Creates a new shallow-copied horroy from another horroy
+ * 
+ * @param {*} value - 
+ * 
+ */
+
+Horroy.from = function (value) {
+    var horr = new Horroy
+
+    horr.length = value.length
+
+    if (typeof value === 'string')
+        for (var i = 0; i < value.length; i++)
+            horr[i] = value[i]
+
+    return horr
+};
+
+Horroy.prototype.push = function () {
+    for (var i = 0; i < arguments.length; i++) {
+        var element = arguments[i];
+        this[this.length++] = element;   
+    }
+    
+    return this.length;
+};
+
+Horroy.prototype.forEach = function (callback) {
+    for (var i = 0; i < this.length; i++)
+        callback(this[i]);
+};
+
+
+Horroy.prototype.fill = function (value, start, end) {
+    if (arguments.length > 4) throw Error('too many arguments');
```

Comment:
You should also validate each parameter type

Reply:
Done! thanks